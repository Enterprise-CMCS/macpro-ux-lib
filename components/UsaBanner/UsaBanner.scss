@use "../../assets/theme/macbis-colors" as c;

// Currently, these colors are only used in the UsaBanner
$black: #1b1b1b;
$gray: #adadad;
$white: #fff;

.usa-banner {
  &.usa-banner--dark {
    background-color: $black;
    .usa-banner__button[aria-expanded="true"]::before {
      background-color: $black;
    }
    .usa-banner__header-text,
    .usa-media-block__body {
      color: $white;
    }
    .usa-banner__button-text,
    .usa-banner__header-action {
      color: $gray;
    }
  }
}
