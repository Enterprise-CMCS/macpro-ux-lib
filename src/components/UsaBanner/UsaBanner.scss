@use "../../assets/theme/macbis-colors" as c;

// TODO: These colors do not exist in _macbis-colors
// How and do we incorporate these? Should these be converted to existing colors?
$black: #1b1b1b;
$gray: #adadad;
$white: #fff;

.usa-banner {
  &.usa-banner--dark {
    background-color: $black;
    .usa-banner__header-text,
    .usa-media-block__body {
      color: $white;
    }
    .usa-banner__button-text {
      color: $gray;
    }
    .usa-banner__button::after {
      background-color: $gray;
    }
  }
}
