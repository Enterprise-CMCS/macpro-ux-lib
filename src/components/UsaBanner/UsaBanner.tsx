import React, { useState } from "react";

import flag from "../../assets/img/us_flag_small.png";
import gov from "../../assets/img/icon-dot-gov.svg";
import lock from "../../assets/img/icon-https.svg";

interface Props {
  className?: string;
  id?: string;
  locale?: "en" | "es";
  variant?: "default" | "dark";
}

/**
 * USA Banner Component
 * @param {string}  className     Additional classes to be added to the root `section` element.
 * @param {string}  id            A unique ID to be applied to the banner content at the root `section` element.
 * @param {string}  locale        Determines which version of the banner to display (en === english, es === español).
 * @param {string}  variant       Indicates which color variant of the banner to display.
 */

export const UsaBanner: React.FC<Props> = ({
  className,
  variant,
  id,
  locale = "en",
  ...rest
}) => {
  const [hidden, setHidden] = useState(true);
  const classNames = `usa-banner${
    variant === "dark" ? ` usa-banner--dark` : ""
  }${className ? ` ${className}` : ""}`;
  return (
    <>
      {locale === "en" && (
        <section
          aria-label="Official government website"
          className={classNames}
          id={id}
          {...rest}
        >
          <div className="usa-accordion">
            <header
              className={`usa-banner__header${
                hidden ? "" : " usa-banner__header--expanded"
              }`}
            >
              <div className="usa-banner__inner">
                <div className="grid-col-auto">
                  <img
                    className="usa-banner__header-flag"
                    src={flag}
                    alt="U.S. flag"
                  />
                </div>
                <div className="grid-col-fill tablet:grid-col-auto">
                  <p className="usa-banner__header-text">
                    An official website of the United States government
                  </p>
                  <p className="usa-banner__header-action" aria-hidden="true">
                    Here's how you know
                  </p>
                </div>
                <button
                  type="button"
                  className="usa-accordion__button usa-banner__button"
                  aria-expanded={hidden ? "false" : "true"}
                  aria-controls="gov-banner-default-default"
                  onClick={() => setHidden(!hidden)}
                >
                  <span className="usa-banner__button-text">
                    Here's how you know
                  </span>
                </button>
              </div>
            </header>
            <div
              className="usa-banner__content usa-accordion__content"
              id="gov-banner-default-default"
              hidden={hidden}
            >
              <div className="grid-row grid-gap-lg">
                <div className="usa-banner__guidance tablet:grid-col-6">
                  <img
                    className="usa-banner__icon usa-media-block__img"
                    src={gov}
                    role="img"
                    alt=""
                    aria-hidden="true"
                  />
                  <div className="usa-media-block__body">
                    <p>
                      <strong>Official websites use .gov</strong>
                      <br />A <strong>.gov</strong> website belongs to an
                      official government organization in the United States.
                    </p>
                  </div>
                </div>
                <div className="usa-banner__guidance tablet:grid-col-6">
                  <img
                    className="usa-banner__icon usa-media-block__img"
                    src={lock}
                    role="img"
                    alt=""
                    aria-hidden="true"
                  />
                  <div className="usa-media-block__body">
                    <p>
                      <strong>Secure .gov websites use HTTPS</strong>
                      <br />A <strong>lock</strong> (
                      <span className="icon-lock">
                        <svg
                          xmlns="http://www.w3.org/2000/svg"
                          width="52"
                          height="64"
                          viewBox="0 0 52 64"
                          className="usa-banner__lock-image"
                          role="img"
                          aria-labelledby="banner-lock-title-default banner-lock-description-default"
                          focusable="false"
                        >
                          <title id="banner-lock-title-default">Lock</title>
                          <desc id="banner-lock-description-default">
                            A locked padlock
                          </desc>
                          <path
                            fill="#000000"
                            fillRule="evenodd"
                            d="M26 0c10.493 0 19 8.507 19 19v9h3a4 4 0 0 1 4 4v28a4 4 0 0 1-4 4H4a4 4 0 0 1-4-4V32a4 4 0 0 1 4-4h3v-9C7 8.507 15.507 0 26 0zm0 8c-5.979 0-10.843 4.77-10.996 10.712L15 19v9h22v-9c0-6.075-4.925-11-11-11z"
                          />
                        </svg>
                      </span>
                      ) or <strong>https://</strong> means you've safely
                      connected to the .gov website. Share sensitive information
                      only on official, secure websites.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      )}
      {locale === "es" && (
        <section
          aria-label="Un sitio oficial del Gobierno de Estados Unidos"
          className={classNames}
          id={id}
          {...rest}
        >
          <div className="usa-accordion">
            <header
              className={`usa-banner__header${
                hidden ? "" : " usa-banner__header--expanded"
              }`}
            >
              <div className="usa-banner__inner">
                <div className="grid-col-auto">
                  <img
                    className="usa-banner__header-flag"
                    src={flag}
                    alt="U.S. flag"
                  />
                </div>
                <div className="grid-col-fill tablet:grid-col-auto">
                  <p className="usa-banner__header-text">
                    Un sitio oficial del Gobierno de Estados Unidos
                  </p>
                  <p className="usa-banner__header-action" aria-hidden="true">
                    Así es como usted puede verificarlo
                  </p>
                </div>
                <button
                  type="button"
                  className="usa-accordion__button usa-banner__button"
                  aria-expanded={hidden ? "false" : "true"}
                  aria-controls="gov-banner-spanish-lang-es"
                  onClick={() => setHidden(!hidden)}
                >
                  <span className="usa-banner__button-text">
                    Así es como usted puede verificarlo
                  </span>
                </button>
              </div>
            </header>
            <div
              className="usa-banner__content usa-accordion__content"
              id="gov-banner-spanish-lang-es"
              hidden={hidden}
            >
              <div className="grid-row grid-gap-lg">
                <div className="usa-banner__guidance tablet:grid-col-6">
                  <img
                    className="usa-banner__icon usa-media-block__img"
                    src={gov}
                    role="img"
                    alt=""
                    aria-hidden="true"
                  />
                  <div className="usa-media-block__body">
                    <p>
                      <strong>Los sitios web oficiales usan .gov</strong>
                      <br />
                      Un sitio web <strong>.gov</strong> pertenece a una
                      organización oficial del Gobierno de Estados Unidos.
                    </p>
                  </div>
                </div>
                <div className="usa-banner__guidance tablet:grid-col-6">
                  <img
                    className="usa-banner__icon usa-media-block__img"
                    src={lock}
                    role="img"
                    alt=""
                    aria-hidden="true"
                  />
                  <div className="usa-media-block__body">
                    <p>
                      <strong>Los sitios web seguros .gov usan HTTPS</strong>
                      <br />
                      Un <strong>candado</strong> (
                      <span className="icon-lock">
                        <svg
                          xmlns="http://www.w3.org/2000/svg"
                          width="52"
                          height="64"
                          viewBox="0 0 52 64"
                          className="usa-banner__lock-image"
                          role="img"
                          aria-labelledby="banner-lock-title-spanish banner-lock-description-spanish"
                          focusable="false"
                        >
                          <title id="banner-lock-title-spanish">Lock</title>
                          <desc id="banner-lock-description-spanish">
                            A locked padlock
                          </desc>
                          <path
                            fill="#000000"
                            fillRule="evenodd"
                            d="M26 0c10.493 0 19 8.507 19 19v9h3a4 4 0 0 1 4 4v28a4 4 0 0 1-4 4H4a4 4 0 0 1-4-4V32a4 4 0 0 1 4-4h3v-9C7 8.507 15.507 0 26 0zm0 8c-5.979 0-10.843 4.77-10.996 10.712L15 19v9h22v-9c0-6.075-4.925-11-11-11z"
                          />
                        </svg>
                      </span>
                      ) o <strong>https://</strong> significa que usted se
                      conectó de forma segura a un sitio web .gov. Comparta
                      información sensible sólo en sitios web oficiales y
                      seguros.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      )}
    </>
  );
};
