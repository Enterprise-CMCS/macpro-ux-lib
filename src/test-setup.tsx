import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

// Here we can set all providers for themes, stores and other stuff
const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <>{children}</>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
