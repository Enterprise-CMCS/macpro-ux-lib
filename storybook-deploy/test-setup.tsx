import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

// Here we can set all providers for themes, stores and other stuff
const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <>{children}</>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };

/**
 * Clean up attributes from the rendered component by removing the generated ids.
 *
 * Before:
 *   for="input-type-text-123456"
 *   id="input-type-text-123456"
 *
 * After:
 *   for="input-type-text"
 *   id="input-type-text"
 *
 * This is useful during snapshot testing. Snapshot comparisons will fail without addressing these items.
 *
 * @param {HTMLElement} container The component to clean up.
 * @param {prefixes}      string[]  The reference substrings to leave alone.
 */
