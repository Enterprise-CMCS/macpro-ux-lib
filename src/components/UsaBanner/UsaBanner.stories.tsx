import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { USWDSDecorator } from "../../utils";
import { UsaBanner } from "./UsaBanner";

export default {
  title: "COMPONENTS/USA Banner",
  component: UsaBanner,
  decorators: [...USWDSDecorator],
  args: {},
  argTypes: {
    className: {
      description:
        "Additional classes to be added to the root `section` element.",
    },
    id: {
      description:
        "A unique ID to be applied to the banner content at the root `section` element.",
    },
    locale: {
      description:
        "Determines which version of the banner to display (en === english, es === espa√±ol).",
    },
  },
} as ComponentMeta<typeof UsaBanner>;

const Template: ComponentStory<typeof UsaBanner> = ({ ...rest }) => (
  <UsaBanner {...rest} />
);

export const Default = Template.bind({});
Default.args = {};

export const EnglishBanner = Template.bind({});
EnglishBanner.args = {
  locale: "en",
};

export const SpanishBanner = Template.bind({});
SpanishBanner.args = {
  locale: "es",
};
