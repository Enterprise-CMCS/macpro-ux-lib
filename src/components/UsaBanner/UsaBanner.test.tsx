import React from "react";
import { UsaBanner } from "./UsaBanner";
import { fireEvent, screen, render } from "../../test-setup";

describe("Tests for the Banner component", () => {
  it("should render", () => {
    // should be English by default
    render(<UsaBanner />);
    expect(
      screen.getByText("An official website of the United States government")
    ).toBeVisible();
  });

  it("should render in english", () => {
    render(<UsaBanner locale="en" />);
    expect(
      screen.getByText("An official website of the United States government")
    ).toBeVisible();
  });

  it("should render in spanish", () => {
    render(<UsaBanner locale="es" />);
    expect(
      screen.getByText("Un sitio oficial del Gobierno de Estados Unidos")
    ).toBeVisible();
  });

  it("should render english with custom class", () => {
    const { container } = render(<UsaBanner className="test-class" />);
    const section = container.firstChild as HTMLElement;
    expect(section.classList.contains("test-class")).toBe(true);
  });

  it("should render spanish with custom class", () => {
    const { container } = render(
      <UsaBanner locale="es" className="test-class" />
    );
    const section = container.firstChild as HTMLElement;
    expect(section.classList.contains("test-class")).toBe(true);
  });

  it("light/default variant", () => {
    const { container } = render(<UsaBanner />);
    const section = container.firstChild as HTMLElement;
    expect(section.className).toBe("usa-banner");
  });

  it("dark variant", () => {
    const { container } = render(<UsaBanner variant="dark" />);
    const section = container.firstChild as HTMLElement;
    expect(section.className).toBe("usa-banner usa-banner--dark");
  });
});

describe("Snapshot tests for UsaBanner component", () => {
  it("English banner", () => {
    const { container } = render(<UsaBanner />);

    // Confirm collapsed
    const content = document.getElementById("gov-banner-default-default");
    expect(content).not.toBeVisible();
    expect(container).toMatchSnapshot(); // Match collapsed state

    // Expand
    const btn = container.getElementsByClassName("usa-banner__button")[0];
    fireEvent.click(btn);
    expect(content).toBeVisible();
    expect(container).toMatchSnapshot(); // Match expanded state
  });

  it("Spanish banner", () => {
    const { container } = render(<UsaBanner locale="es" />);

    // Confirm collapsed
    const content = document.getElementById("gov-banner-spanish-lang-es");
    expect(content).not.toBeVisible();
    expect(container).toMatchSnapshot(); // Match collapsed state

    // Expand
    const btn = container.getElementsByClassName("usa-banner__button")[0];
    fireEvent.click(btn);
    expect(content).toBeVisible();
    expect(container).toMatchSnapshot(); // Match expanded state
  });
});
