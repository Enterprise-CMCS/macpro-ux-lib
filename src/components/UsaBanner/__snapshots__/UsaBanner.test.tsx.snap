// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for UsaBanner component English banner 1`] = `
<div>
  <section
    aria-label="Official government website"
    class="usa-banner"
  >
    <div
      class="usa-accordion"
    >
      <header
        class="usa-banner__header"
      >
        <div
          class="usa-banner__inner"
        >
          <div
            class="grid-col-auto"
          >
            <img
              alt="U.S. flag"
              class="usa-banner__header-flag"
              src="/assets/img/us_flag_small.png"
            />
          </div>
          <div
            class="grid-col-fill tablet:grid-col-auto"
          >
            <p
              class="usa-banner__header-text"
            >
              An official website of the United States government
            </p>
            <p
              aria-hidden="true"
              class="usa-banner__header-action"
            >
              Here's how you know
            </p>
          </div>
          <button
            aria-controls="gov-banner-default-default"
            aria-expanded="false"
            class="usa-accordion__button usa-banner__button"
            type="button"
          >
            <span
              class="usa-banner__button-text"
            >
              Here's how you know
            </span>
          </button>
        </div>
      </header>
      <div
        class="usa-banner__content usa-accordion__content"
        hidden=""
        id="gov-banner-default-default"
      >
        <div
          class="grid-row grid-gap-lg"
        >
          <div
            class="usa-banner__guidance tablet:grid-col-6"
          >
            <img
              alt=""
              aria-hidden="true"
              class="usa-banner__icon usa-media-block__img"
              role="img"
              src="/assets/img/icon-dot-gov.svg"
            />
            <div
              class="usa-media-block__body"
            >
              <p>
                <strong>
                  Official websites use .gov
                </strong>
                <br />
                A 
                <strong>
                  .gov
                </strong>
                 website belongs to an official government organization in the United States.
              </p>
            </div>
          </div>
          <div
            class="usa-banner__guidance tablet:grid-col-6"
          >
            <img
              alt=""
              aria-hidden="true"
              class="usa-banner__icon usa-media-block__img"
              role="img"
              src="/assets/img/icon-https.svg"
            />
            <div
              class="usa-media-block__body"
            >
              <p>
                <strong>
                  Secure .gov websites use HTTPS
                </strong>
                <br />
                A 
                <strong>
                  lock
                </strong>
                 (
                <span
                  class="icon-lock"
                >
                  <svg
                    aria-labelledby="banner-lock-title-default banner-lock-description-default"
                    class="usa-banner__lock-image"
                    focusable="false"
                    height="64"
                    role="img"
                    viewBox="0 0 52 64"
                    width="52"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="banner-lock-title-default"
                    >
                      Lock
                    </title>
                    <desc
                      id="banner-lock-description-default"
                    >
                      A locked padlock
                    </desc>
                    <path
                      d="M26 0c10.493 0 19 8.507 19 19v9h3a4 4 0 0 1 4 4v28a4 4 0 0 1-4 4H4a4 4 0 0 1-4-4V32a4 4 0 0 1 4-4h3v-9C7 8.507 15.507 0 26 0zm0 8c-5.979 0-10.843 4.77-10.996 10.712L15 19v9h22v-9c0-6.075-4.925-11-11-11z"
                      fill="#000000"
                      fill-rule="evenodd"
                    />
                  </svg>
                   
                </span>
                ) or 
                <strong>
                  https://
                </strong>
                 means you've safely connected to the .gov website. Share sensitive information only on official, secure websites.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Snapshot tests for UsaBanner component English banner 2`] = `
<div>
  <section
    aria-label="Official government website"
    class="usa-banner"
  >
    <div
      class="usa-accordion"
    >
      <header
        class="usa-banner__header usa-banner__header--expanded"
      >
        <div
          class="usa-banner__inner"
        >
          <div
            class="grid-col-auto"
          >
            <img
              alt="U.S. flag"
              class="usa-banner__header-flag"
              src="/assets/img/us_flag_small.png"
            />
          </div>
          <div
            class="grid-col-fill tablet:grid-col-auto"
          >
            <p
              class="usa-banner__header-text"
            >
              An official website of the United States government
            </p>
            <p
              aria-hidden="true"
              class="usa-banner__header-action"
            >
              Here's how you know
            </p>
          </div>
          <button
            aria-controls="gov-banner-default-default"
            aria-expanded="true"
            class="usa-accordion__button usa-banner__button"
            type="button"
          >
            <span
              class="usa-banner__button-text"
            >
              Here's how you know
            </span>
          </button>
        </div>
      </header>
      <div
        class="usa-banner__content usa-accordion__content"
        id="gov-banner-default-default"
      >
        <div
          class="grid-row grid-gap-lg"
        >
          <div
            class="usa-banner__guidance tablet:grid-col-6"
          >
            <img
              alt=""
              aria-hidden="true"
              class="usa-banner__icon usa-media-block__img"
              role="img"
              src="/assets/img/icon-dot-gov.svg"
            />
            <div
              class="usa-media-block__body"
            >
              <p>
                <strong>
                  Official websites use .gov
                </strong>
                <br />
                A 
                <strong>
                  .gov
                </strong>
                 website belongs to an official government organization in the United States.
              </p>
            </div>
          </div>
          <div
            class="usa-banner__guidance tablet:grid-col-6"
          >
            <img
              alt=""
              aria-hidden="true"
              class="usa-banner__icon usa-media-block__img"
              role="img"
              src="/assets/img/icon-https.svg"
            />
            <div
              class="usa-media-block__body"
            >
              <p>
                <strong>
                  Secure .gov websites use HTTPS
                </strong>
                <br />
                A 
                <strong>
                  lock
                </strong>
                 (
                <span
                  class="icon-lock"
                >
                  <svg
                    aria-labelledby="banner-lock-title-default banner-lock-description-default"
                    class="usa-banner__lock-image"
                    focusable="false"
                    height="64"
                    role="img"
                    viewBox="0 0 52 64"
                    width="52"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="banner-lock-title-default"
                    >
                      Lock
                    </title>
                    <desc
                      id="banner-lock-description-default"
                    >
                      A locked padlock
                    </desc>
                    <path
                      d="M26 0c10.493 0 19 8.507 19 19v9h3a4 4 0 0 1 4 4v28a4 4 0 0 1-4 4H4a4 4 0 0 1-4-4V32a4 4 0 0 1 4-4h3v-9C7 8.507 15.507 0 26 0zm0 8c-5.979 0-10.843 4.77-10.996 10.712L15 19v9h22v-9c0-6.075-4.925-11-11-11z"
                      fill="#000000"
                      fill-rule="evenodd"
                    />
                  </svg>
                   
                </span>
                ) or 
                <strong>
                  https://
                </strong>
                 means you've safely connected to the .gov website. Share sensitive information only on official, secure websites.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Snapshot tests for UsaBanner component Spanish banner 1`] = `
<div>
  <section
    aria-label="Un sitio oficial del Gobierno de Estados Unidos"
    class="usa-banner"
  >
    <div
      class="usa-accordion"
    >
      <header
        class="usa-banner__header"
      >
        <div
          class="usa-banner__inner"
        >
          <div
            class="grid-col-auto"
          >
            <img
              alt="U.S. flag"
              class="usa-banner__header-flag"
              src="/assets/img/us_flag_small.png"
            />
          </div>
          <div
            class="grid-col-fill tablet:grid-col-auto"
          >
            <p
              class="usa-banner__header-text"
            >
              Un sitio oficial del Gobierno de Estados Unidos
            </p>
            <p
              aria-hidden="true"
              class="usa-banner__header-action"
            >
              Así es como usted puede verificarlo
            </p>
          </div>
          <button
            aria-controls="gov-banner-spanish-lang-es"
            aria-expanded="false"
            class="usa-accordion__button usa-banner__button"
            type="button"
          >
            <span
              class="usa-banner__button-text"
            >
              Así es como usted puede verificarlo
            </span>
          </button>
        </div>
      </header>
      <div
        class="usa-banner__content usa-accordion__content"
        hidden=""
        id="gov-banner-spanish-lang-es"
      >
        <div
          class="grid-row grid-gap-lg"
        >
          <div
            class="usa-banner__guidance tablet:grid-col-6"
          >
            <img
              alt=""
              aria-hidden="true"
              class="usa-banner__icon usa-media-block__img"
              role="img"
              src="/assets/img/icon-dot-gov.svg"
            />
            <div
              class="usa-media-block__body"
            >
              <p>
                <strong>
                  Los sitios web oficiales usan .gov
                </strong>
                <br />
                Un sitio web 
                <strong>
                  .gov
                </strong>
                 pertenece a una organización oficial del Gobierno de Estados Unidos.
              </p>
            </div>
          </div>
          <div
            class="usa-banner__guidance tablet:grid-col-6"
          >
            <img
              alt=""
              aria-hidden="true"
              class="usa-banner__icon usa-media-block__img"
              role="img"
              src="/assets/img/icon-https.svg"
            />
            <div
              class="usa-media-block__body"
            >
              <p>
                <strong>
                  Los sitios web seguros .gov usan HTTPS
                </strong>
                <br />
                Un 
                <strong>
                  candado
                </strong>
                 (
                <span
                  class="icon-lock"
                >
                  <svg
                    aria-labelledby="banner-lock-title-spanish banner-lock-description-spanish"
                    class="usa-banner__lock-image"
                    focusable="false"
                    height="64"
                    role="img"
                    viewBox="0 0 52 64"
                    width="52"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="banner-lock-title-spanish"
                    >
                      Lock
                    </title>
                    <desc
                      id="banner-lock-description-spanish"
                    >
                      A locked padlock
                    </desc>
                    <path
                      d="M26 0c10.493 0 19 8.507 19 19v9h3a4 4 0 0 1 4 4v28a4 4 0 0 1-4 4H4a4 4 0 0 1-4-4V32a4 4 0 0 1 4-4h3v-9C7 8.507 15.507 0 26 0zm0 8c-5.979 0-10.843 4.77-10.996 10.712L15 19v9h22v-9c0-6.075-4.925-11-11-11z"
                      fill="#000000"
                      fill-rule="evenodd"
                    />
                  </svg>
                   
                </span>
                ) o 
                <strong>
                  https://
                </strong>
                 significa que usted se conectó de forma segura a un sitio web .gov. Comparta información sensible sólo en sitios web oficiales y seguros.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Snapshot tests for UsaBanner component Spanish banner 2`] = `
<div>
  <section
    aria-label="Un sitio oficial del Gobierno de Estados Unidos"
    class="usa-banner"
  >
    <div
      class="usa-accordion"
    >
      <header
        class="usa-banner__header usa-banner__header--expanded"
      >
        <div
          class="usa-banner__inner"
        >
          <div
            class="grid-col-auto"
          >
            <img
              alt="U.S. flag"
              class="usa-banner__header-flag"
              src="/assets/img/us_flag_small.png"
            />
          </div>
          <div
            class="grid-col-fill tablet:grid-col-auto"
          >
            <p
              class="usa-banner__header-text"
            >
              Un sitio oficial del Gobierno de Estados Unidos
            </p>
            <p
              aria-hidden="true"
              class="usa-banner__header-action"
            >
              Así es como usted puede verificarlo
            </p>
          </div>
          <button
            aria-controls="gov-banner-spanish-lang-es"
            aria-expanded="true"
            class="usa-accordion__button usa-banner__button"
            type="button"
          >
            <span
              class="usa-banner__button-text"
            >
              Así es como usted puede verificarlo
            </span>
          </button>
        </div>
      </header>
      <div
        class="usa-banner__content usa-accordion__content"
        id="gov-banner-spanish-lang-es"
      >
        <div
          class="grid-row grid-gap-lg"
        >
          <div
            class="usa-banner__guidance tablet:grid-col-6"
          >
            <img
              alt=""
              aria-hidden="true"
              class="usa-banner__icon usa-media-block__img"
              role="img"
              src="/assets/img/icon-dot-gov.svg"
            />
            <div
              class="usa-media-block__body"
            >
              <p>
                <strong>
                  Los sitios web oficiales usan .gov
                </strong>
                <br />
                Un sitio web 
                <strong>
                  .gov
                </strong>
                 pertenece a una organización oficial del Gobierno de Estados Unidos.
              </p>
            </div>
          </div>
          <div
            class="usa-banner__guidance tablet:grid-col-6"
          >
            <img
              alt=""
              aria-hidden="true"
              class="usa-banner__icon usa-media-block__img"
              role="img"
              src="/assets/img/icon-https.svg"
            />
            <div
              class="usa-media-block__body"
            >
              <p>
                <strong>
                  Los sitios web seguros .gov usan HTTPS
                </strong>
                <br />
                Un 
                <strong>
                  candado
                </strong>
                 (
                <span
                  class="icon-lock"
                >
                  <svg
                    aria-labelledby="banner-lock-title-spanish banner-lock-description-spanish"
                    class="usa-banner__lock-image"
                    focusable="false"
                    height="64"
                    role="img"
                    viewBox="0 0 52 64"
                    width="52"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="banner-lock-title-spanish"
                    >
                      Lock
                    </title>
                    <desc
                      id="banner-lock-description-spanish"
                    >
                      A locked padlock
                    </desc>
                    <path
                      d="M26 0c10.493 0 19 8.507 19 19v9h3a4 4 0 0 1 4 4v28a4 4 0 0 1-4 4H4a4 4 0 0 1-4-4V32a4 4 0 0 1 4-4h3v-9C7 8.507 15.507 0 26 0zm0 8c-5.979 0-10.843 4.77-10.996 10.712L15 19v9h22v-9c0-6.075-4.925-11-11-11z"
                      fill="#000000"
                      fill-rule="evenodd"
                    />
                  </svg>
                   
                </span>
                ) o 
                <strong>
                  https://
                </strong>
                 significa que usted se conectó de forma segura a un sitio web .gov. Comparta información sensible sólo en sitios web oficiales y seguros.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
